tests/test_meta.py

```diff
@@ -0,0 +1,39 @@
+from pathlib import Path
+
+import pytest
+
+VS16 = b"\xef\xb8\x8f"
+VS16_SUPPRESS_COMMENT = b"# test: allow-vs16"
+
+
+def check_dir_for_vs16(path: Path) -> None:
+    error_locations: list[tuple[Path, int]] = []
+    for file in path.rglob("*.py"):
+        for line_no, line in enumerate(file.read_bytes().splitlines(), 1):
+            if VS16 in line and VS16_SUPPRESS_COMMENT not in line:
+                error_locations.append((file, line_no))
+
+    if error_locations:
+        raise AssertionError(
+            f"unsuppressed VS16 found in {len(error_locations)} lines:\n"
+            + "\n".join(f"- {file}:{line_no}" for file, line_no in error_locations)
+        )
+
+
+def test_src_unwanted_vs16() -> None:
+    check_dir_for_vs16(Path(__file__).parents[1] / "app")
```

Comment:
This should probably check the tests too, since any test that compares emojis will probably get confused by variation selectors.
```suggestion
    src_dir = Path(__file__).parents[1]
    check_dir_for_vs16(src_dir / "app")
    check_dir_for_vs16(src_dir / "tests")
```